micro-rdk/src/common/credentials_storage.rs

```diff
@@ -90,12 +92,18 @@ pub struct TlsCertificate {
     pub(crate) private_key: Vec<u8>,
 }
 
-impl From<CertificateResponse> for TlsCertificate {
-    fn from(resp: CertificateResponse) -> Self {
-        Self {
-            certificate: resp.tls_certificate.into_bytes(),
-            private_key: resp.tls_private_key.into_bytes(),
-        }
+impl TryFrom<CertificateResponse> for TlsCertificate {
+    type Error = rustls_pki_types::pem::Error;
+    fn try_from(resp: CertificateResponse) -> Result<Self, Self::Error> {
+        // we convert the certificate and private key from PEM to DER format to save space
+        let private_key_bytes = resp.tls_private_key.into_bytes();
+        let private_key = PrivateKeyDer::from_pem_slice(&private_key_bytes[0..])?;
```

Comment:
this range should just work with `[..]` right? also, maybe check if the range is needed or if `as_slice` is available.
L102 does the same